src/vm/ceemain.cpp

```diff
@@ -1718,6 +1718,7 @@ void STDMETHODCALLTYPE EEShutDownHelper(BOOL fIsDllUnloading)
                     {
                         IBCLoggingDisabler disableLogging( pInfo );  // runs IBCLoggingDisabler::DisableLogging
                         
+                        CONTRACT_VIOLATION(GCViolation);
```

Comment:
This function has the `GC_TRIGGERS` as does `WriteAllModuleProfileData()` or does that not have anything to do with `GCViolation`?

Reply:
`CrstHolder lock(g_IBCLogger.GetSync());` is switching to `GC_NOTRIGGER`.

Reply:
Oh. I thought that the contract is scoped and didn't escape the scope in which it was set. I am also assuming you are referring to the below at the beginning of the `GetSync()` function or do I need to go deeper to where the switch is taking place?

>    CONTRACTL
>    {
>        THROWS;
>        GC_NOTRIGGER;
>        MODE_ANY;
>        SO_NOT_MAINLINE;
>    }
>    CONTRACTL_END;

Reply:
Ah... there are two classes named `CrstHolder` 😞 One in `crst.h` and one in `gcenv.sync.h`. I think I should be looking in `crst.h` I guess.